src/main/webapp/app/text-shared/highlighted-text-area/highlighted-text-area.component.ts

```diff
@@ -46,7 +48,14 @@ export class HighlightedTextAreaComponent implements OnChanges, DoCheck {
             if (assessment.reference == null) {
                 return content;
             }
-            return content.replace(assessment.reference, `<span class="highlight ${HighlightColors.forIndex(currentIndex)}">${assessment.reference}</span>`);
+
+            let replacementString = assessment.reference;
```

Comment:
this could also be solved with a logical or:

`let replacementString = this.blocks[currentIndex] || assessment.reference`

Also for someone who doesn't know that much about the text submission process, I don't understand what this distinction means, so a comment would be nice.

Reply:
I would love to do it like that, however this does not allow for something like this:
`let replacementString = this.blocks[currentIndex]`**`.text`**` || assessment.reference`

However, I think I found a better way to express it and also upped comments for explanation.